### PR TITLE
Changes was added by Oleh Kashetskyi

### PR DESCRIPTION
**Pull request by appsterlight2k**
Please check code

### CODE REVIEW

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/config/HelperWordsProperties.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import java.util.Set;
+
+@Getter
+@Setter
+@Component
+@ConfigurationProperties(prefix = "helper")
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/config/HelperWordsProperties.java:13]: yep, another way to get properties, nice

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/facory/NotificationServiceFactory.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package co.inventorsoft.academy.spring.facory;
+
+@Component
+public class NotificationServiceFactory {
+
+    @Autowired
+    private EmailNotificationService emailNotificationService;
+
+    @Autowired
+    private SlackNotificationService slackNotificationService;
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/facory/NotificationServiceFactory.java]: prefer constructor injection

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/ArticleRepository.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        InputStream is = getClass().getResourceAsStream("/articles.json");
+
+        Optional<InputStream> isOpt = Optional.ofNullable(is);
+        if (isOpt.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Type articlesListType = new TypeToken<ArrayList<Article>>() {}.getType();
+
+        return new Gson().fromJson(new InputStreamReader(isOpt.get()), articlesListType);
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/ArticleRepository.java]: inject gson bean, do not create it each time

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/CategoryRepository.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        InputStream is = getClass().getResourceAsStream("/categories.json");
+
+        Optional<InputStream> isOpt = Optional.ofNullable(is);
+        if (isOpt.isEmpty()) {
+            return Collections.emptySet();
+        }
+
+        Type categoriesType = new TypeToken<Set<String>>() {}.getType();
+
+        return new Gson().fromJson(new InputStreamReader(isOpt.get()), categoriesType);
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/CategoryRepository.java]: inject gson

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/UserRepository.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        InputStream is = getClass().getResourceAsStream("/users.json");
+
+        Optional<InputStream> isOpt = Optional.ofNullable(is);
+        if (isOpt.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        Type userListType = new TypeToken<ArrayList<User>>() {}.getType();
+
+        return new Gson().fromJson(new InputStreamReader(isOpt.get()), userListType);
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/UserRepository.java]: same

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryFileService.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Set;
+
+@Service
+public class CategoryFileService {
+    public void saveCategories(Set<String> categories) {
+        File file = new File("article-analyzer/src/main/resources/categories.json");
+        Gson gson = new Gson();
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryFileService.java]: inject gson

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+public class CategoryService {
+    @Autowired
+    private HelperWordsProperties helperWordsProperties;
+    @Autowired
+    private ArticleRepository articleRepository;
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java]: prefer constructor injection

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+    public Set<String> extractCategories() {
```
Comment (vitaliiStefanchakInventorsoft) [inline article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java:28]: nice method

**Review by vitaliiStefanchakInventorsoft — COMMENTED** [review]
nice work, just few minor comments

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/facory/NotificationServiceFactory.java (hunk 2 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/facory/NotificationServiceFactory.java]: Fixed

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/ArticleRepository.java (hunk 3 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/ArticleRepository.java]: Fixed

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/CategoryRepository.java (hunk 4 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/CategoryRepository.java]: Done

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/UserRepository.java (hunk 5 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/repository/UserRepository.java]: Done

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryFileService.java (hunk 6 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryFileService.java]: Fixed

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java (hunk 7 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/service/CategoryService.java]: Refactored, done!

File: article-analyzer/src/main/java/co/inventorsoft/academy/spring/config/HelperWordsProperties.java (hunk 1 above)
Comment (appsterlight2k) [reply article-analyzer/src/main/java/co/inventorsoft/academy/spring/config/HelperWordsProperties.java:13]: Heah, it's a little bit more convenient

**Review by vitaliiStefanchakInventorsoft — APPROVED** [review]

**Comment by vitaliiStefanchakInventorsoft** [comment]
ConsiderItDoneItsDoneGIF

**Comment by appsterlight2k** [comment]
Thank you for your cooperation)

**Pull request closed without merging**